### PR TITLE
Add Xamarin Forms sample project

### PR DESCRIPTION
**Pull request by cabauman**
- Utilizes offline and authentication features.

### CODE REVIEW

**Comment by bezysoftware** [comment]
Hi, thanks for the PR, I pretty busy lately, but hope to take a look by the end of the week

**Comment by cabauman** [comment]
Sure, I understand. No rush. Love what you've done with this project.

**Comment by cabauman** [comment]
Okay. I tried with my Redmi physical device and a Pixel 2 emulator, and wasn't able to reproduce the exception. I'll try again a bit later.

File: samples/XamarinForms/XamarinForms/Services/FirebaseOfflineDataStore.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            {
+                OfflineDatabaseFactory = (t, s) => new OfflineDatabase(t, s),
+                AuthTokenAsyncFactory = async () => await authService.GetFirebaseAuthToken()
+            };
+
+            // The offline database filename is named after type T.
+            // So, if you have more than one list of type T objects, you need to differentiate it
+            // by adding a filename modifier; which is what we're using the "key" parameter for.
+            var client = new FirebaseClient(BaseUrl, options);
+            _realtimeDb = client
```
[the remark this answers is not in the document]
Comment (cabauman) [reply samples/XamarinForms/XamarinForms/Services/FirebaseOfflineDataStore.cs]: The problem is Xamarin Live Player. It's _still_ in preview and has [a number of limitations](https://docs.microsoft.com/en-us/xamarin/tools/live-player/limitations). But if you run on a real device or emulator everything works fine.

The exception occurs [here](https://github.com/step-up-labs/firebase-database-dotnet/blob/d87d17015606c872b1c336af813e77e97b925810/src/Firebase/Offline/OfflineDatabase.cs#L33) so it's probably LiteDB's [use of expresions](https://github.com/mbdavid/LiteDB/blob/0864d8f888af427571487b84652567c1fd78d97f/LiteDB/Mapper/EntityBuilder.cs#L50) that Live Player doesn't like.

**Pull request merged**